Cleanup the repo a little bit after Rust Reference decision

JoelMarcey opened the pull request:
There is still more to do, but this tries to at least reduce the confusion that this repo is actively being worked on at this point in time.

JoelMarcey committed:
Cleanup the repo a little bit after Rust Reference decision

There is still more two do, but this tries to at least reduce the confusion that this repo is actively being worked on at this point in time.

JoelMarcey committed:
Forget to add content to a file

JoelMarcey commented:
What do I need to change/update so `mdbook build` doesn't try to build deleted files any longer?

ehuss commented:
> What do I need to change/update so `mdbook build` doesn't try to build deleted files any longer?

I think for now we should just keep the `spec` directory. It is the only way to test the `mdbook-spec` plugin to ensure that it works on a basic level.

The plan is to move it to the reference repo once the workspace problems have been worked out in rust-lang/rust. At that time, we can delete most everything here.

If you'd like, perhaps change this PR to only modify the top-level README, and we can take care of the rest soon afterwards?

ehuss committed:
Remove unused .gitignore file

This was missed in https://github.com/rust-lang/spec/pull/64.

ehuss committed:
Remove mdbook-spec

This has moved to the reference repo in https://github.com/rust-lang/reference/pull/1542.

ehuss committed:
Remove spec source README.

There isn't anything here, and no intent to add anything, and there
shouldn't be any links to this.

ehuss committed:
Remove authoring guide

This has moved to https://github.com/rust-lang/reference/blob/master/docs/authoring.md

ehuss committed:
Remove the book

The book text is now in https://github.com/rust-lang/reference/

ehuss committed:
Update CONTRIBUTING now that the book is not here

ehuss committed:
Remove workflows

These are no longer needed now that mdbook-spec and the book are not
here.

ehuss commented:
I pushed some commits to further clean up the repo after https://github.com/rust-lang/reference/pull/1542.
